tests/components/scrape/test_init.py

```diff
@@ -45,15 +53,25 @@ async def test_setup_no_data_fails(hass: HomeAssistant) -> None:
         ]
     }
 
+    mocker = MockRestData("test_scrape_sensor_no_data")
     with patch(
-        "homeassistant.components.scrape.coordinator.RestData",
-        return_value=MockRestData("test_scrape_sensor_no_data"),
+        "homeassistant.components.rest.RestData",
+        return_value=mocker,
     ):
-        assert not await async_setup_component(hass, DOMAIN, config)
+        assert await async_setup_component(hass, DOMAIN, config)
+        await hass.async_block_till_done()
+
+        state = hass.states.get("sensor.ha_version")
+        assert state is None
+
+        assert "Platform scrape not ready yet" in caplog.text
+
+        mocker.payload = "test_scrape_sensor"
```

Comment:
Maybe better to leave the patch context on line 64, and create a new one here with a new mocker.

Reply:
(Maybe or maybe not here - Martin has more experience with tests so maybe this is ok)

Reply:
It's fine to change the mock attributes on the fly as long as there's a new call or access after that, that we want to assert.